### PR TITLE
helm-ls 0.0.9 (new formula)

### PR DESCRIPTION
**Pull request by qvalentin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/h/helm-ls.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class HelmLs < Formula
+  desc "Language server for Helm"
+  homepage "https://github.com/mrjosh/helm-ls"
+  url "https://github.com/mrjosh/helm-ls.git",
+      tag:      "v0.0.9",
+      revision: "5dbde8a5d83a7cd3c21e55e8150100a2f7dd462e"
```
Comment (p-linnane) [inline Formula/h/helm-ls.rb:6]:
```suggestion
  url "https://github.com/mrjosh/helm-ls/archive/refs/tags/v0.0.9.tar.gz"
  sha256 "cef59537d2335c9821a2d60628bcf40451a137b9f5b38b3991d1914c3f4ec191"
```

Comment (p-linnane) [reply Formula/h/helm-ls.rb:6]: Let's use a tarball to avoid having to clone the repo.

File: Formula/h/helm-ls.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  license "MIT"
+  head "https://github.com/mrjosh/helm-ls.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    system "make"
+    bin.install "bin/helm_ls"
```
Comment (p-linnane) [inline Formula/h/helm-ls.rb:14]:
```suggestion
    ldflags = %W[
      -X main.Version=#{version}
      -X main.CompiledBy=#{tap.user}
      -X main.GitCommit=#{tap.user}
      -X main.BuildTime=#{time.iso8601}
    ]
    system "go", "build", *std_go_args(ldflags: ldflags, output: bin/"helm_ls")
```

File: Formula/h/helm-ls.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    generate_completions_from_executable(bin/"helm_ls", "completion")
+  end
+
+  test do
+    system bin/"helm_ls", "version"
```
Comment (p-linnane) [inline Formula/h/helm-ls.rb:20]:
```suggestion
    assert_match version.to_s, shell_output(bin/"helm_ls version")
```

Comment (p-linnane) [reply Formula/h/helm-ls.rb:20]: We will need an additional test that exercises some functionality of the software. Versions or help checks rarely fail even if there are compilation issues. In this case it's worth keeping to ensure the ldflags are set correctly.

**Review by p-linnane — CHANGES_REQUESTED** [review]
Thanks for your submission. I have a few changes I'd like to see prior to merging, as well as the addition of another functional test.

**Comment by qvalentin** [comment]
Thanks for the suggestions.

File: Formula/h/helm-ls.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    cwd = Dir.pwd
+    json = <<~JSON
+      {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "method": "initialize",
+        "params": {
+          "workspaceFolders": [
+            {
+              "uri": "file://#{cwd}"
```
Comment (chenrui333) [inline Formula/h/helm-ls.rb:37]:
```suggestion
    json = <<~JSON
      {
        "jsonrpc": "2.0",
        "id": 1,
        "method": "initialize",
        "params": {
          "workspaceFolders": [
            {
              "uri": "file://#{testpath}"
```

File: Formula/h/helm-ls.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      -X main.GitCommit=#{tap.user}
+      -X main.BuildTime=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags, output: bin/"helm_ls")
+
+    generate_completions_from_executable(bin/"helm_ls", "completion")
+  end
+
+  test do
+    require 'open3'
```
Comment (chenrui333) [inline Formula/h/helm-ls.rb:24]: some style issue
```suggestion
    require "open3"
```

**Review by SMillerDev — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

**Pull request merged**